packages/text-editor/src/components/extension/validateContent.ts

```diff
@@ -0,0 +1,43 @@
+import { Editor, Extension } from '@tiptap/core'
+
+export interface ValidateContentOptions {
+  onEmpty?: (isEmpty: boolean) => void
+  onValidate?: (html: string) => void
+}
+
+export interface ValidateContentStorage {
+  isEmpty: boolean
+}
+
+const handleChange = (editor: Editor, options: ValidateContentOptions, storage: ValidateContentStorage): void => {
+  if (options.onEmpty !== undefined && options.onEmpty !== null) {
+    if (storage.isEmpty !== editor.isEmpty) {
+      storage.isEmpty = editor.isEmpty
```

Comment:
Is `editor.isEmpty` = `true` for `<p></p>` value in the editor?

Reply:
It depends on the schema. It checks whether content matches the default content. If the content is block, then yes, `editor.isEmpty` will be `true` for `<p></p>`

https://github.com/ueberdosis/tiptap/issues/580#issuecomment-826398844
https://github.com/ueberdosis/tiptap/blob/b359f137e9777200e2483b1754ec1987e7d59765/packages/core/src/Editor.ts#L452

Reply:
Yes

Reply:
ok, great